ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js

```diff
@@ -299,11 +302,18 @@ export default class ConfirmApproveContent extends Component {
 
   renderDataContent() {
     const { t } = this.context;
-    const { data } = this.props;
+    const { data, isSetApproveForAll, setApproveForAllArg } = this.props;
     return (
       <div className="flex-column">
         <div className="confirm-approve-content__small-text">
-          {t('functionApprove')}
+          {isSetApproveForAll
+            ? t('functionSetApprovalForAll')
+            : t('functionApprove')}
+        </div>
+        <div className="confirm-approve-content__small-text">
```

Comment:
nit and non-blocker: Could avoid showing an empty element when we are not displaying `isSetApproveForAll`

Reply:
Done here: https://github.com/MetaMask/metamask-extension/commit/2584852a0987a52666319cdfef6d4f1cbdb97c93